### PR TITLE
RHICf Run and Event number added in StRHICfCollection

### PR DESCRIPTION
**Pull request by ggfdsa10**
PR 1. StRHICfCollection for RHICf Run and Event Number

I will request 4 PRs for adding the RHICf run and event number, before the new STAR release 
1) this PR
2) StRHICfDbMaker
3) StMuDstMaker/StRHICfRawHit and StRHICfUtil
4) StRHICfRawHitMaker

I believe reviewing these PRs is very simple and clear.
Please review and approve this 

Thank you, 
Seunghwan

### CODE REVIEW

**Review by plexoos — APPROVED** [review]
LGTM

File: StRoot/StEvent/StRHICfCollection.h [hunk 1]
```diff
@@ -31,13 +31,17 @@ class StRHICfCollection : public StObject
...
+    void setRHICfEventNumber(UInt_t event);
     void setBunchNumber(UInt_t bunch);
     void setRunType(UInt_t type);
     void setTriggerNumber(UInt_t trigger);
     void setRunTime(Int_t idx, UInt_t time);
     void setRunTRGM(UInt_t trgm);
 
     UInt_t numberOfPoints() const;
+    UInt_t getRHICfRunNumber();
+    UInt_t getRHICfEventNumber();
```
Comment (plexoos) [inline StRoot/StEvent/StRHICfCollection.h]:
```suggestion
    UInt_t getRHICfRunNumber() const;
    UInt_t getRHICfEventNumber() const;
```

File: StRoot/StEvent/StRHICfCollection.cxx [hunk 2]
```diff
@@ -58,6 +62,8 @@ void StRHICfCollection::setRunTRGM(UInt_t trgm){mRunTRGM = trgm;}
 
 //=========== Get ===========//
 UInt_t StRHICfCollection::numberOfPoints() const {return mRHICfPointColl.size();}
+UInt_t StRHICfCollection::getRHICfRunNumber(){return mRHICfRunNumber;}
+UInt_t StRHICfCollection::getRHICfEventNumber(){return mRHICfEventNumber;}
```
Comment (plexoos) [inline StRoot/StEvent/StRHICfCollection.cxx]:
```suggestion
UInt_t StRHICfCollection::getRHICfRunNumber() const { return mRHICfRunNumber; }
UInt_t StRHICfCollection::getRHICfEventNumber() const { return mRHICfEventNumber; }
```

**Review by klendathu2k — COMMENTED** [review]
The code is clear and unobjectionable (save for constness which Dmitri has noted).  What is not clear is the rationale.  Why do you need to propagate a run and event number, when these are already provided by StEvent?  Two possible reasons come to mind...

1) Downstream analysis software expects them to be part of the StEvent / StMuDst and/or pico dst objects.  If this is the case, then we _should not_ approve this pull request.

2) The RHICf detector has encoded a run and event numbering scheme that differs from the STAR scheme.   If this is the case, why do we need to have two different run/event numbering schemes in our event data model?  This is guaranteed to cause confusion in the future.

**Comment by genevb** [comment]
> Why do you need to propagate a run and event number, when these are already provided by StEvent? ...
> 2. The RHICf detector has encoded a run and event numbering scheme that differs from the STAR scheme.   If this is the case, why do we need to have two different run/event numbering schemes in our event data model?  This is guaranteed to cause confusion in the future.

To expedite, I will try to summarize my understanding on behalf of the RHICf folks:

#2 is correct, and this is because RHICf did take their own raw data outside of the STAR DAQ system. That raw data has its own runs and events. By including the _RHICf_ run and event numbers in addition to the _STAR_ run and event numbers that are already accessible in the DSTs, the RHICf group can correlate between findings they made through their own raw data investigations with what they get through the STAR framework.

**Comment by klendathu2k** [comment]
> https://github.com/star-bnl/star-sw/pull/2 is correct, and this is because RHICf did take their own raw data outside of the STAR DAQ system.

Thanks Gene.  I thought that would be the case.  Inclusion in the STAR event data model is reasonable,  But it absolutely must be documented in the header file.

File: StRoot/StEvent/StRHICfCollection.cxx [hunk 3]
```diff
@@ -50,6 +52,8 @@ const std::vector<StRHICfPoint*>& StRHICfCollection::pointCollection() const {re
 void StRHICfCollection::isAllSave(){mRHICfHitColl = new StRHICfHit();}
 
 //=========== Set ===========//
+void StRHICfCollection::setRHICfRunNumber(UInt_t run){mRHICfRunNumber = run;}
```
Comment (klendathu2k) [inline StRoot/StEvent/StRHICfCollection.cxx:55]: It is important that we not confuse people looking at the data in the future.  Please provide a comment block for the run and event number setters, getters and member variables.  These comment blocks should clearly state what is being stored/retrieved, and why it differs from the STAR run and event numbering scheme.

File: StRoot/StEvent/StRHICfCollection.h [hunk 4]
```diff
@@ -49,6 +53,8 @@ class StRHICfCollection : public StObject
     StRHICfHit* mRHICfHitColl; 
     std::vector<StRHICfPoint*> mRHICfPointColl; 
 
+    UInt_t mRHICfRunNumber;
```
Comment (genevb) [inline StRoot/StEvent/StRHICfCollection.h:58]:
```suggestion
    // Important note: the _RHICf_ Run and Event numbers are distinct from the _STAR_ Run and Event numbers,
    // originating from RHICf's own raw data acquisition and studies outside of the STAR framework
    UInt_t mRHICfRunNumber;
```

**Comment by ggfdsa10** [comment]
Dear Jason,
Thank you for your comments 

and Gene, 
Thank you for your explanation on behalf of RHICf and me

**Review by plexoos — APPROVED** [review]

**Review by klendathu2k — APPROVED** [review]
LGTM

**Pull request merged**